studio/pages/cli/login.tsx

```diff
@@ -0,0 +1,58 @@
+import { NextPage } from 'next'
+import APIAuthorizationLayout from 'components/layouts/APIAuthorizationLayout'
+import { createCliLoginSession } from 'data/cli/login'
+import { useParams } from 'common'
+import { useRouter } from 'next/router'
+import { withAuth } from 'hooks'
+import Connecting from 'components/ui/Loading/Loading'
+import { useEffect, useState } from 'react'
+
+const CliLoginPage: NextPage = () => {
+  const router = useRouter()
+  const { session_id, public_key, token_name, success } = useParams()
+  const [isSuccessfulLogin, setSuccessfulLogin] = useState(false)
+
+  useEffect(() => {
+    if (!router.isReady) {
+      return
+    }
+
+    if (success) {
+      setSuccessfulLogin(true)
+      return
+    }
+
+    async function createSession() {
+      if (!session_id || !public_key) {
+        router.push('/404')
+        return
+      }
+
+      const session = await createCliLoginSession(session_id, public_key, token_name)
```

Comment:
should wrap this in a try catch and handle accordingly? since createCliLoginSession could throw an error

catch could just show a toast.error(...)